### PR TITLE
endpoint: download file

### PR DESCRIPTION
**Pull request by parthc2fo**
-  I have added a download endpoint to start downloading the file
-  URL would be: `${url}/download/${fileId}`

### CODE REVIEW

File: src/download/download.service.ts [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+   * @param fileId unique fileId
+   */
+  async downloadFile(res: Response, fileId: string): Promise<void> {
+    try {
+      const fileObj = await this.fileRepository.findOne(fileId);
+
+      switch (fileObj?.uploadingStatus) {
+        case UPLOAD_TYPE.COMPLETED:
+          const path = `${BUCKET_DIRECTORY}/${fileObj.filenameOnBucket}`;
+          res.download(path);
```
Comment (yadavpradeep786) [inline src/download/download.service.ts:31]: This will only work when file is open to everyone. Read file buffer and download the same on FE

**Review by yadavpradeep786 — CHANGES_REQUESTED** [review]

File: src/download/download.service.ts (hunk 1 above)
Comment (parthc2fo) [reply src/download/download.service.ts:31]: Can you please explain  meaning of  the file is open to everyone here?

- Read the file buffer and download the same on FE 
 ->  I think internally it is working the same. it is actually streaming a file.  
 
 Link: https://github.com/expressjs/express/blob/master/lib/response.js#L1042

Comment (yadavpradeep786) [reply src/download/download.service.ts:31]: LGTM

**Review by yadavpradeep786 — APPROVED** [review]
LGTM

**Comment by parthc2fo** [comment]
I am merging this branch

**Pull request merged**